trieste/models/gpflow/models.py

```diff
@@ -507,13 +496,291 @@ def conditional_predict_y(
+    """
+
+    def __init__(
+        self,
+        model: SGPR,
+        optimizer: Optimizer | None = None,
+        num_rff_features: int = 1000,
+        inducing_point_selector: Optional[
+            InducingPointSelector[SparseGaussianProcessRegression]
+        ] = None,
+    ):
+        """
+        :param model: The GPflow model to wrap.
+        :param optimizer: The optimizer with which to train the model. Defaults to
+            :class:`~trieste.models.optimizer.Optimizer` with :class:`~gpflow.optimizers.Scipy`.
+        :param num_rff_features: The number of random Fourier features used to approximate the
+            kernel when calling :meth:`trajectory_sampler`. We use a default of 1000 as it
+            typically perfoms well for a wide range of kernels. Note that very smooth
+            kernels (e.g. RBF) can be well-approximated with fewer features.
+        :param inducing_point_selector: The (optional) desired inducing_point_selector that
+            will update the underlying GPflow SGPR model's inducing points as
+            the optimization progresses.
+        :raise NotImplementedError: If we try to use an inducing_point_selector with a model
+            that has more than one set of inducing points.
+        """
+        super().__init__(optimizer)
+        self._model = model
+
+        check_optimizer(self.optimizer)
+
+        if num_rff_features <= 0:
+            raise ValueError(
+                f"num_rff_features must be greater or equal to zero but got {num_rff_features}."
+            )
+        self._num_rff_features = num_rff_features
+
+        if isinstance(self.model.inducing_variable, SeparateIndependentInducingVariables):
+            if inducing_point_selector is not None:
+                raise NotImplementedError(
+                    f"""
+                    InducingPointSelectors only currently support models with a single set
+                    of inducing points however received inducing points of
+                    type {type(self.model.inducing_variable)}.
+                    """
+                )
+        self._inducing_point_selector = inducing_point_selector
+
+        self._ensure_variable_model_data()
+        self.create_posterior_cache()
+
+    def __repr__(self) -> str:
+        """"""
+        return (
+            f"SparseGaussianProcessRegression({self.model!r}, {self.optimizer!r},"
+            f"{self._num_rff_features!r}, {self._inducing_point_selector!r})"
+        )
+
+    @property
+    def model(self) -> SGPR:
+        return self._model
+
+    @property
+    def inducing_point_selector(
+        self,
+    ) -> Optional[InducingPointSelector[SparseGaussianProcessRegression]]:
+        return self._inducing_point_selector
+
+    def predict_y(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        f_mean, f_var = self.predict(query_points)
+        return self.model.likelihood.predict_mean_and_var(f_mean, f_var)
```

Comment:
no cacheing?

Reply:
done automatically by GPflowPredictor now in `predict` call, that's why we need to reimplement `predict_y` method here